tiledb/sm/query/result_tile.cc

```diff
@@ -446,45 +446,8 @@ inline uint8_t ResultTile::str_coord_intersects(
     const char* const buff_str,
     const std::string& range_start,
     const std::string& range_end) {
-  // Test against start
-  bool geq_start = true;
-  bool all_chars_match = true;
-  uint64_t min_size = std::min<uint64_t>(c_size, range_start.size());
-  for (uint64_t i = 0; i < min_size; ++i) {
-    if (buff_str[c_offset + i] < range_start[i]) {
-      geq_start = false;
-      all_chars_match = false;
-      break;
-    } else if (buff_str[c_offset + i] > range_start[i]) {
-      all_chars_match = false;
-      break;
-    }  // Else characters match
-  }
-  if (geq_start && all_chars_match && c_size < range_start.size()) {
-    geq_start = false;
-  }
-
-  // Test against end
-  bool seq_end = false;
-  if (geq_start) {
-    seq_end = true;
-    all_chars_match = true;
-    min_size = std::min<uint64_t>(c_size, range_end.size());
-    for (uint64_t i = 0; i < min_size; ++i) {
-      if (buff_str[c_offset + i] > range_end[i]) {
-        geq_start = false;
-        break;
-      } else if (buff_str[c_offset + i] < range_end[i]) {
-        all_chars_match = false;
-        break;
-      }  // Else characters match
-    }
-    if (seq_end && all_chars_match && c_size > range_end.size()) {
-      seq_end = false;
-    }
-  }
-
-  return geq_start && seq_end;
+  auto str = std::string(buff_str + c_offset, c_size);
```

Comment:
I suggest being sure your (performance) test set includes coordinate sets that fall outside the 'internal' storage size of any string implementation we use to see how it does with the heap allocation/deallocation that will be involved, and involving multiple threads (IIRC there are situations where this may be simultaneously in use from multiple threads, thus with possibility of heap contention for those alloc/dealloc.)

Reply:
Also, used to be possible to have range_end.empty(), and if that is still possible then your current comparison will probably fail for some data.

Reply:
@eric-hughes-tiledb I saw you use this in the overlap function. Maybe you have some insight on performance here?

Reply:
I'm unclear about which "overlap function" you're referring to. The `overlap_ratio` function I worked on a few months ago was for integers and floating point types, not strings. If you're referring to `Interval<T>::intersection`, it uses `operator<` (or related) defined by the class `T`.

On general grounds, I'd expect that string comparison function built in to the standard library to be at least as good as custom code for comparison. The compiler vendor has good reason to make those comparison well optimized.

Reply:
And, I should say, having looked at this code, it would be worth taking the time to declare it returning `bool` as it should have from the start.

Reply:
@eric-hughes-tiledb I think the concern here is the added creation of a std::string on the stack. Any thoughts?

Reply:
Ah. That has an easy solution. The goal is to use a standard-library comparison function. To get that, you don't need to convert to `std::string`, but only to `std::string_view`. The strings are allocated for the lifespan of the function call, so they don't need to be copied or allocated (what `string` would provide). All that's needed is to let the library see the strings in a way that it expects through `string_view`.

Reply:
thanks @eric-hughes-tiledb!